packages/react-native-executorch/common/rnexecutorch/data_processing/dsp.cpp

```diff
@@ -0,0 +1,65 @@
+#include <algorithm>
+#include <cstddef>
+#include <limits>
+#include <math.h>
+#include <rnexecutorch/data_processing/FFT.h>
+#include <rnexecutorch/data_processing/dsp.h>
+
+namespace rnexecutorch::dsp {
+
+using namespace rnexecutorch::dsp;
+
+std::vector<float> hannWindow(int size) {
+  // https://www.mathworks.com/help/signal/ref/hann.html
+  std::vector<float> window(size);
+  for (int i = 0; i < size; i++) {
+    window[i] = 0.5f * (1 - std::cos(2 * M_PI * i / size));
+  }
+  return window;
+}
+
+std::vector<float> stftFromWaveform(std::span<float> waveform,
+                                    size_t fftWindowSize, size_t hopSize) {
+  // Initialize FFT
+  FFT fft(fftWindowSize);
```

Comment:
prevent implicit conversion from size_t to int since FFT constructor accepts int
```suggestion
  FFT fft(static_cast<int>(fftWindowSize));
```